service/client_state.cc

```diff
@@ -244,6 +253,23 @@ future<bool> service::client_state::check_has_permission(auth::command_desc cmd)
     co_return false;
 }
 
+future<bool> service::client_state::check_has_permissions(auth::command_desc cmd, const auth::permission_set& perms) const {
```

Comment:
I see 3 problems here:
- name of the function doesn't hint if this is ALL or ANY check.
- code duplication could be avoided
- it's not ok that in some cases we validate on cmd.permission and other on perms arg.

Reply:
Or create different command type with set, and overload checking permissions function, or template it. There are many solutions but we have 2 constrains:
- no perf degradation
- code is not confusing or duplicated